src/mscorlib/src/System/BitConverter.cs

```diff
@@ -339,21 +339,25 @@ public static bool ToBoolean(byte[] value, int startIndex)
             return value[startIndex] != 0;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)] // This should compile down to nothing or roughly 1 instruction, depending on the register holding value
```

Comment:
It seems like the JIT isn't optimizing the last case quite correctly, as it could be generating:
```asm
sub         rsp,28h
vzeroupper
mov         rcx,7FFFFFFFFFFFFFFFh              ; bits = *((long*)&value) & 0x7FFFFFFFFFFFFFFF
and         rcx,qword ptr [7FFD435F04D0h]      ;
mov         rax,7FF0000000000000h              ; result = bits > 0x7FF0000000000000
cmp         rcx,rax                            ;
setg        cl                                 ;
movzx       ecx,cl
```